spieleplattform-frontend/src/pages/Game/GamePage/StarComponent/StarComponent.tsx

```diff
@@ -0,0 +1,18 @@
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import "./StarComponent.scss"
+
+export default function StarComponent(starProps: {count: number}) {
+
+    const stars = []
+    for(let index = 0; index < starProps.count; index++) {
+        stars.push(
+            <FontAwesomeIcon icon={["fas", "star"]} size={"2x"} key={index}/>
+        )
+    }
```

Comment:
Ich bin immer ein großer Fan von Higher Order functions und versuche for loops, wann immer es geht zu vermeiden. In vielen Fällen wird der Code dadurch kürzer und lesbarer und du musst keinen State mutieren. In diesem Fall ist das nicht ganz so schlimm da du deine stars Variable nur vorrübergehend zur instanziierung mutierst. Das hier wäre zum Beispiel eine Möglichkeit das ganze mit higher order function zu schreiben: `Array.from({ length: starProps.count }, (_, i) => <FontAwesomeIcon icon={["fas", "star"]} size={"2x"} key={i} />)`